### PR TITLE
💄 📦 use fontawesome icons in AppBar

### PR DESCRIPTION
**Pull request by jdhoffa**
- 📦 gains fontawesome dependency
- use fontawesome icons instead of buttons

## Before
Screenshot 2024-12-03 at 15 24 35

## After
Screenshot 2024-12-03 at 15 27 36

Towards #91

### CODE REVIEW

**Comment by jdhoffa** [comment]
I will wait for @MonikaFu 's review here, as this PR adds a new dependency (and also is a totally superfluous style change, so not urgent)

**Comment by MonikaFu** [comment]
Not sure if it is a good change. Maybe for GitHub it is but not sure if the save is clear enough as an icon to be honest.

**Comment by jdhoffa** [comment]
> Not sure if it is a good change. Maybe for GitHub it is but not sure if the save is clear enough as an icon to be honest.

Yup totally fair enough. I thought it looked more elegant than the buttons, but perhaps also less usable. 

With that said, I imagine if a user was curious they would just click the icon, and relatively quickly understand that it saves a data_archive ¯\_(ツ)_/¯

**Comment by jdhoffa** [comment]
I do feel that the icons look more professional though...

## Before
Screenshot 2024-12-04 at 22 32 18

## After
Screenshot 2024-12-04 at 22 31 40

**Comment by jdhoffa** [comment]
I added a hover effect which could be a compromise (although I'm also not convinced of this):
Screenshot 2024-12-05 at 12 29 40

(Sorry the screenshot is sort of bogus)

**Comment by jdhoffa** [comment]
@MonikaFu FYI, a call with @hodie lead to the following feedback: 

- The icons look more professional than the buttons
- The buttons/ text is more user-friendly than the icons
- Both features in question (GitHub, downloading an archive of the data) are likely reserved for pretty advanced users either way

A general comment was also made: 
These icons probably shouldn't be in the AppBar, but rather in a footer of the dashboard. They should exist and be available to advanced users who know what they want/ know that the buttons are there, but don't need to be front and center.

Overall actions then are:
- [ ] Review/ merge this PR
- [ ] Probably open a new issue/ PR to move icons to the footer of the page (not critically important right now)

**Comment by hodie** [comment]
> @MonikaFu FYI, a call with @hodie lead to the following feedback:
> 
> * The icons look more professional than the buttons
> * The buttons/ text is more user-friendly than the icons
> * Both features in question (GitHub, downloading an archive of the data) are likely reserved for pretty advanced users either way
> 
> A general comment was also made: These icons probably shouldn't be in the AppBar, but rather in a footer of the dashboard. They should exist and be available to advanced users who know what they want/ know that the buttons are there, but don't need to be front and center.
> 
> Overall actions then are:
> 
> * [ ]  Review/ merge this PR
> * [x]  Probably open a new issue/ PR to move icons to the footer of the page (not critically important right now)
>   
>   * Tracked in [🎨 move advanced features to a less dominant part of the dashboard #135](https://github.com/RMI-PACTA/pacta-dashboard-svelte/issues/135)

Just one clarification: the conclusion was to have icons. If/when we have a proper bottom bar (like e.g. here on GitHub) there should be a text for GitHub just as there should be for Terms, Security, Contact. The download action probably does make sense to have on top since it's an action about the content. But the whole top bar will need to be reviewed once we can test it end-to-end and the dashboard is embedded into another page. Seems like a top-bar and especially the RMI logo will be confusing in that context.

**Comment by jdhoffa** [comment]
FYI @MonikaFu I removed the hover over effect cause it was clunky and distracted, but this is ready for review now.

File: src/routes/+layout.svelte [hunk 1]
```diff
@@ -60,18 +60,29 @@
...
-				<button class="btn btn-sm variant-filled-primary" on:click={downloadArchive}
-					>Save Data</button
-				>
-				<button class="btn btn-sm variant-filled-primary">Help</button>
+				<span class="btn-icon variant-filled">
+					<a
+						href="https://github.com/RMI-PACTA/pacta-dashboard-svelte"
+						target="_blank"
+						rel="noreferrer"
+					>
```
Comment (MonikaFu) [inline src/routes/+layout.svelte:68]:
```suggestion
title="GitHub"
					>
```

File: src/routes/+layout.svelte [hunk 2]
```diff
@@ -60,18 +60,29 @@
...
+						<i class="fab fa-xl fa-github"></i>
+					</a>
+				</span>
+				<span class="btn-icon variant-filled">
+					<button on:click={downloadArchive}>
```
Comment (MonikaFu) [inline src/routes/+layout.svelte:73]:
```suggestion
					<button on:click={downloadArchive} title="Save data">
```

File: src/routes/+layout.svelte [hunk 3]
```diff
@@ -60,18 +60,29 @@
...
+						<i class="fas fa-xl fa-floppy-disk"></i>
+					</button>
+					<div class="card p-4 variant-filled-primary" data-popup="popupHover">
+						<p>Save Data</p>
+						<div class="arrow variant-filled-primary" />
+					</div>
+				</span>
+				<span class="btn-icon variant-filled">
+					<a href="https://rmi.gitbook.io/pacta-knowledge-hub" target="_blank" rel="noreferrer">
```
Comment (MonikaFu) [inline src/routes/+layout.svelte:82]:
```suggestion
					<a href="https://rmi.gitbook.io/pacta-knowledge-hub" target="_blank" rel="noreferrer" title="PACTA Knowledge Hub">
```

**Review by MonikaFu — CHANGES_REQUESTED** [review]
Adding a title to the icons results in a small banner appearing when they hover over the icon for longer. I think it adds a bit to the user friendliness without sacrificing the lean look of the icons.

**Review by MonikaFu — APPROVED** [review]
LGTM

**Pull request merged**